pkg/clustercache/clustercache.go

```diff
@@ -77,30 +81,41 @@ type Deployment struct {
 	Name                    string
 	Namespace               string
 	Labels                  map[string]string
+	Annotations             map[string]string
 	MatchLabels             map[string]string
 	SpecSelector            *metav1.LabelSelector
 	SpecReplicas            *int32
+	SpecStrategy            appsv1.DeploymentStrategy
 	StatusAvailableReplicas int32
+	PodSpec                 PodSpec
```

Comment:
I believe spec will have the highest memory footprint. However, it still serves as a good starting point and offers potential for future optimization.

Reply:
Let me merge this and open a new change set to reduce this further!